core/sup_tree_core/executor_pool/memcache_manager.ex

```diff
@@ -0,0 +1,103 @@
+# Copyright(c) 2015-2018 ACCESS CO., LTD. All rights reserved.
+
+use Croma
+
+defmodule AntikytheraCore.ExecutorPool.MemcacheManager do
+  @moduledoc """
+  A `GenServer` to manage memcache for each executor pool.
+  """
+
+  @timeout         300_000
+  @max_records_num 10
+  @max_record_size 65536
+
+  use GenServer
+  alias AntikytheraCore.Ets.Memcache
+  alias AntikytheraCore.ExecutorPool.RegisteredName, as: RegName
+  alias Antikythera.ExecutorPool.Id, as: EPoolId
+
+  def start_link(name, epool_id) do
+    GenServer.start_link(__MODULE__, epool_id, [name: name])
+  end
+
+  @impl true
+  def init(epool_id) do
+    {:ok, %{epool_id: epool_id, records: %{}, expire_at_set: :gb_sets.new()}, @timeout}
+  end
+
+  @impl true
+  def handle_call({:write, key, value, lifetime, ratio}, _from, state) do
+    evicted_state = evict_expired_records(state, System.monotonic_time(:milliseconds))
+    if :erts_debug.flat_size(value) > @max_record_size do
+      {:reply, {:error, :too_large_object}, evicted_state, @timeout}
+    else
+      new_state =
+        evicted_state
+        |> evict_head_record_if_exceeds_limit()
```

Comment:
We shouldn't evict the record at head when the key already exists in the cache (in this case number of records don't change).
Let's call `evict_head_record_if_exceeds_limit` from within `do_write/5`.

Reply:
How about calling `evict_head_record_if_exceeds_limit` after `do_write`?
It's more concise but exceeds the limit in a moment.

Reply:
Yeah, changing order of the function calls suffices. Go ahead!

Reply:
Fixed in 6ae7fc15a013142f6442c519f27080b81a63bbc9